source/common.ts

```diff
@@ -0,0 +1,51 @@
+import { createOwnCloudFileDetailProperty } from './helper';
```

Comment:
Were there no intentions to make `nextcloud-link` more functional/monadic at/from the start? (@floriansimon1?)

Reply:
I don't exactly understand where the question comes from, but I can tell you was to have the convenience of functions (no state based on this) with the convenience of grouping functions on an object.

Reply:
Well I think simply from the realization that, for instance, there aren't any monads used in nextcloud-link. (Be that lonad, folktale, or anything else). And to understand whether that was by conscious choice for a specific reason (keep number of external dependencies low, etc)

Reply:
I definitely wanted the dependencies count to a minimum